google-cloud-bigtable/lib/google/cloud/bigtable/instance.rb

```diff
@@ -0,0 +1,432 @@
+        # creation process encounters an error.
+        #
+        # @return [Boolean]
+
+        def creating?
+          state == :CREATING
+        end
+
+        # Instance type. Possible values are `:DEVELOPMENT`, `:PRODUCTION`,
+        # `:TYPE_UNSPECIFIED`
+        #
+        # @return [Symbol]
+
+        def type
+          @grpc.type
+        end
+
+        # The instance is meant for development and testing purposes only; it has
+        # no performance or uptime guarantees and is not covered by SLA.
+        # After a development instance is created, it can be upgraded by
+        # updating the instance to type `PRODUCTION`. An instance created
+        # as a production instance cannot be changed to a development instance.
+        # When creating a development instance, `serve_nodes` on the cluster must
+        # not be set.
+        #
+        # @return [Boolean]
+
+        def development?
+          type == :DEVELOPMENT
+        end
+
+        # An instance meant for production use. `serve_nodes` must be set
+        # on the cluster.
+        #
+        # @return [Boolean]
+
+        def production?
+          type == :PRODUCTION
+        end
+
+        # Set instance type.
+        #
+        # Valid values are `:DEVELOPMENT`, `:PRODUCTION`.
+        # If instance already created then After a development instance is
+        # created, it can be upgraded by updating the instance to type `PRODUCTION`.
+        # An instance created as a production instance cannot be changed to a
+        # development instance.
+        #
+        # @param instance_type [Symbol]
+
+        def type= instance_type
+          @grpc.type = instance_type
+        end
+
+        # Get instance labels.
+        #
+        # Cloud Labels are a flexible and lightweight mechanism for organizing
+        # cloud resources into groups that reflect a customer's organizational
+        # needs and deployment strategies. Cloud Labels can be used to filter
+        # collections of resources. They can be used to control how resource
+        # metrics are aggregated. And they can be used as arguments to policy
+        # management rules (e.g. route, firewall, load balancing, etc.).
+        #
+        # * Label keys must be between 1 and 63 characters long and must conform
+        #   to the following regular expression: `[a-z]([-a-z0-9]*[a-z0-9])?`.
+        # * Label values must be between 0 and 63 characters long and must
+        #   conform to the regular expression `([a-z]([-a-z0-9]*[a-z0-9])?)?`.
+        # * No more than 64 labels can be associated with a given resource.
+        #
+        # @return [Hash{String=>String}] The label keys and values in a hash.
+
+        def labels
+          @grpc.labels
+        end
+
+        # Updates the Cloud Labels.
+        #
+        # @param labels [Hash{String=>String}] The Cloud Labels.
+
+        def labels= labels
+          @grpc.labels = Google::Protobuf::Map.new(
```

Comment:
What if users pass in `nil`?

Reply:
I will add type check with allowed only hash type argument.

Reply:
IMO `nil` should be a supported type. The protobuf object can have a map set to `nil`.

Reply:
Got it. For nil value, I think need to clear labels because instance gRPC object is throwing an error "TypeError: Expected Map instance".